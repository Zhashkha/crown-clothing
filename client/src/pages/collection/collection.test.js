import React from "react";
import { shallow } from "enzyme";

import { CollectionPage } from "./collection.component";
import CollectionItem from "../../components/collection-item/collection-item.component";

describe("CollectionPage", () => {
  let wrapper;
  let mockItems = [{ id: 1, id: 2, id: 3 }];
  beforeEach(() => {
    const mockProps = {
      collection: { title: "", items: mockItems }
    };

    wrapper = shallow(<CollectionPage {...mockProps} />);
  });

  it("test CollectionPage snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the same number of CollectionItem components as in mock array", () => {
    expect(wrapper.find(CollectionItem).length).toBe(mockItems.length);
  });
});
