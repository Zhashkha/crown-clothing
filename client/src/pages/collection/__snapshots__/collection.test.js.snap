// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPage test CollectionPage snapshot 1`] = `ShallowWrapper {}`;
